service/qos/service_level_controller.hh

```diff
@@ -259,6 +259,10 @@ public:
         return sl_it->second;
     }
 
+    bool has_service_level(const sstring& service_level_name) const {
```

Comment:
Nit (for the future): prefer `std::string_view` to `const sstring&`. C++20's containers support heterogenous lookup, so using it in `.contains` should not result in a temporary copy being constructed.

Fortunately (for the sake of merging the PR now), `has_service_level` right now is only used in conjunction with a `sstring` parameter so there would be no performance gain.